### PR TITLE
Improve home assistant auto discovery

### PR DESCRIPTION
**Pull request by cybe**
# Description

Hey,

after starting using @hobbypunk90 auto discovery integration (Thank you!) I noticed a bug: Using a device topic prefix with a slash leads to that slash being preserved when [forming the topic](https://github.com/zewelor/bt-mqtt-gateway/blob/8fbf3540bd8058ad9054fc4a3d4552420865335e/workers/mithermometer.py#L34) where the config is being send to.

So a topic prefix of `foo/bar/baz` leads to something like: `homeassistant/sensor/foo/bar/baz_thermometer_bedroom_temperature/config` and is thus violating the [discovery protocol](https://www.home-assistant.io/docs/mqtt/discovery/).

While fixing the bug I also also started to restructure a few bits here and there (inspired by zigbee2mqtt):
* Group multiple sensors of a device under a common [`node_id`](https://www.home-assistant.io/docs/mqtt/discovery/) (that is the mac address).
* The `object_id` is being constructed from the name of the module, the device and the sensor.
* Clearly differentiate when generating various IDs, names and topics. 
* Use the discovery name as a technical ID. One can set a so called friendly name within home assistant itself if so desired.
* Use some constants to refer to common sensor values
* Added various `unit_of_measurement`s

An example for a mithermometer device:

Topics:
  * `{discovery_prefix}/{sensor_type}/{node_id}/{object_id}/config`
  * `my_discovery_prefix/sensor/4C:65:A8:DA:44:A2/mithermometer_wohnen_temperature/config`
  
Configuration:
```
{
  "unique_id": "bt-mqtt-gateway/4C:65:A8:DA:44:A2/mithermometer_wohnen_temperature",
  "name": "mithermometer_wohnen_temperature",
  "state_topic": "my_prefix/bt-mqtt-gateway/mithermometer/wohnen/temperature",
  "device_class": "temperature",
  "device": {
    "identifiers": [
      "4C:65:A8:D5:E4:E1",
      "bt-mqtt-gateway/4C:65:A8:DA:44:A2/mithermometer_wohnen"
    ],
    "manufacturer": "Xiaomi",
    "model": "LYWSD(CGQ/01ZM)",
    "name": "mithermometer_wohnen"
  }
}
```

What do you guys think?

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by cybe** [comment]
Also @hobbypunk90:
Are [these payload definitions](https://github.com/zewelor/bt-mqtt-gateway/blob/8dfda3080dd1b53c0f7c45bf7637ddc79a2c6287/workers/thermostat.py#L94-L95) with single quotes intentional? They seem to result in Home Assistant errors:`2019-06-28 13:14:42 ERROR (MainThread) [custom_components.mqtt.climate] Invalid _away mode: off`. Removing those single quotes fixes these errors.

**Comment by hobbypunk90** [comment]
@cybe i think you are right, removing the single quotes should be ok 🤔

File: workers/base.py [hunk 1]
```diff
@@ -8,11 +8,19 @@ def __init__(self, command_timeout, **args):
   def _setup(self):
     return
 
-  def format_id(self, *args, separator='/'):
-    return separator.join(["bt-mqtt-gateway", repr(self), *args])
+  def format_discovery_topic(self, mac, *sensor_args):
+    node_id = '0x{}'.format(mac.replace(':', ''))
```
Comment (hobbypunk90) [inline workers/base.py]: is it really needed? i think the mac as the node_id should be ok, or not (node_id = mac)?
i'm not sure if its ok for mqtt thinking if not, i think node_id = mac.replace(':', '-') is a more readable solution.

Comment (cybe) [reply workers/base.py]: mqtt allows for either `-` or `:`.The idea was to keep it short. But we might just use node_id = mac, so: `my_discovery_prefix/sensor/4C:65:A8:DA:44:A2/mithermometer_wohnen_temperature` as an example.

Comment (ronytomen) [reply workers/base.py]: I use full mac addresses in my implementations with MQTT as a "node" identifier.  Easy to understand what you are looking at and intuitive for those who are familiar with MAC addresses and what they mean for BLE.

Comment (cybe) [reply workers/base.py]: I agree, @ronytomen.

**Comment by cybe** [comment]
I changed the node_id to use the full mac address and edited the description. It's ready to merge from my point of view.

**Comment by zewelor** [comment]
Looks great thanks !

**Pull request merged**